core/pkg/evaluator/json.go

```diff
@@ -301,22 +314,21 @@ func resolve[T constraints](reqID string, key string, context map[string]any, va
 	return value, variant, reason, metadata, nil
 }
 
-// runs the rules (if defined) to determine the variant, otherwise falling through to the default
 // nolint: funlen
-func (je *JSON) evaluateVariant(reqID string, flagKey string, context map[string]any) (
+func (je *Resolver) evaluateVariant(reqID string, flagKey string, c map[string]any) (
 	variant string, variants map[string]interface{}, reason string, metadata map[string]interface{}, err error,
 ) {
 	metadata = map[string]interface{}{}
 
-	flag, ok := je.store.Get(flagKey)
+	flag, ok := je.store.Get(context.Background(), flagKey)
```

Comment:
maybe we can create a `ctx := context.Background()` and reuse it in other calls

Reply:
@thisthat good suggestion, updated with commit https://github.com/open-feature/flagd/pull/1259/commits/ec1c907e3016b51886ea72938007f51c2b749685